### PR TITLE
Add support for the secret scanning API

### LINKED ISSUE #2278: Support Secret Scanning endpoints
Add support to retrieve and update secret scanning alerts for an enterprise, organization or repository ([Secret Scanning API reference](https://docs.github.com/en/rest/reference/secret-scanning#list-secret-scanning-alerts-for-a-repository)).

This is useful to automate tasks around security analysis and reporting.

### PR DESCRIPTION
**Pull request by crqra**
This PR adds support for the [Secret Scanning API](https://docs.github.com/en/rest/reference/secret-scanning). Includes all endpoints and structures listed on the API documentation.

Closes #2278

### CODE REVIEW

File: github/secret_scanning.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+	LocationsURL *string    `json:"locations_url,omitempty"`
+	State        *string    `json:"state,omitempty"`
+	Resolution   *string    `json:"resolution,omitempty"`
+	ResolvedAt   *Timestamp `json:"resolved_at,omitempty"`
+	ResolvedBy   *User      `json:"resolved_by,omitempty"`
+	SecretType   *string    `json:"secret_type,omitempty"`
+	Secret       *string    `json:"secret,omitempty"`
+}
+
+type SecretScanningAlertLocation struct {
```
Comment (gmlewis) [inline github/secret_scanning.go:33]: and this one

File: github/secret_scanning.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+	ListCursorOptions
+}
+
+// SecretScanningAlertLocationListOptions specifies optional parameters to the SecretScanningService.ListLocationsForAlert method.
+type SecretScanningAlertLocationListOptions struct {
+	ListOptions
+}
+
+// SecretScanningAlertUpdateOptions specifies optional parameters to the SecretScanningService.UpdateAlert method.
```
Comment (gmlewis) [inline github/secret_scanning.go:70]: Let's please get rid of this and reuse the one above.

File: github/secret_scanning.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+	// A comma-separated list of resolutions. Only secret scanning alerts with one of these resolutions are listed.
+	// Valid resolutions are false_positive, wont_fix, revoked, pattern_edited, pattern_deleted or used_in_tests.
+	Resolution string `url:"resolution,omitempty"`
+
+	ListCursorOptions
+}
+
+// SecretScanningAlertLocationListOptions specifies optional parameters to the SecretScanningService.ListLocationsForAlert method.
+type SecretScanningAlertLocationListOptions struct {
```
Comment (gmlewis) [inline github/secret_scanning.go:66]: Let's please get rid of this and just use `ListOptions`.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @crqra !

File: github/secret_scanning.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+	Type    *string                             `json:"type,omitempty"`
+	Details *SecretScanningAlertLocationDetails `json:"details,omitempty"`
+}
+
+type SecretScanningAlertLocationDetails struct {
+	Path        *string `json:"path,omitempty"`
+	Startline   *int64  `json:"start_line,omitempty"`
+	EndLine     *int64  `json:"end_line,omitempty"`
+	StartColumn *int64  `json:"start_column,omitempty"`
+	EndColumn   *int64  `json:"end_column,omitempty"`
```
[the remark this answers is not in the document]
Comment (crqra) [reply github/secret_scanning.go:43]: Makes sense 👍🏼

File: github/secret_scanning.go (hunk 2 above)
Comment (crqra) [reply github/secret_scanning.go:70]: Although these structs are very similar, the use cases and therefore the documentation is quite different.

Should I copy the documentation over to the `SecretScanningAlertListOptions`, making it clear the values have different meanings depending on where they're being used?

Comment (gmlewis) [reply github/secret_scanning.go:68]: Oh, OK... Then that is fine if you wish to leave these as-they-are.

**Comment by crqra** [comment]
Thanks for the feedback @gmlewis! I've updated the PR with your suggestions, LMK if it looks good 😄

**Review by gmlewis — APPROVED** [review]
Thank you, @crqra !
LGTM.
Awaiting second LGTM/Approval before merging.

**Comment by gmlewis** [comment]
@Parker77 - do you have time to review this PR?

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
Thank you, @Parker77 !
Merging.

**Pull request merged**